tdms/src/input_matrices.cpp

```diff
@@ -0,0 +1,140 @@
+#include "input_matrices.h"
+
+#include <string.h>
+#include <stdexcept>
+
+#include <spdlog/spdlog.h>
+
+#include "fdtd_grid_initialiser.h"
+#include "utils.h"
+#include "matlabio.h"
+
+using namespace std;
+using namespace tdms_matrix_names;
+
+int index_from_matrix_name(const char *matrix_name) {
+    for (int i = 0; i < NMATRICES; i++) {
+        if(are_equal(matrix_name, matrixnames[i])) { return i; }
+    }
+    throw runtime_error("Matrix name " + string(matrix_name) + " not found in matrixnames.");
+}
```

Comment:
An `std::vector` can be searched with [`std::find`](https://en.cppreference.com/w/cpp/algorithm/find)... doesn't make the code spectacularly more beautiful but _you_ don't need to play with string equality.

```c++
int index_from_matrix_name(const std::string& matrix_name) {
    auto position = std::find(matrixnames.begin(), matrixnames.end(), matrix_name);
    if (position == matrixnames.end()) throw runtime_error(...);
    return std::distance(matrixnames.begin(), position);
}
```